### ISSUE TITLE
.NET core on Linux CENTOS7; remote MS SQL DB; Migrations run only halfway through

### ISSUE THREAD

**Issue by JohnRanger** [volkanceylan/Serenity#4705]
Hi all,

I am trying (just to learn something) to do my 1st SF-based .net core project on a Linux machine.

I have read all docs that I could find and so far have successfully installed a .net core 2.2 SDK, created my 1st serene project on the LX box and have successfully connected it to a fresh MS SQL instance on another machine (Windows).

When I do a dotnet run, the migrations start to execute (the serene and the northwind databases are auto-created within the MS SQL instance on the other box and within the serene DB the usual tables are created. 

But then the execution of the app crashes with an exception around the System.IO.StreamReader that a value cannot be null (probably the ).

It crashes within NorthwindDB_20141123_155100_initial.cs in the following code block:

```cs
private string GetScript(string name)
        {
	  Console.WriteLine("\u001b[41;1m\u001b[41;1mSQL Script:" + name + "\u001b[0m" ); // *** Debug output ***
        
            using (var sr = new StreamReader(this.GetType().Assembly.GetManifestResourceStream(name)))  // <<<<<---- Here it crashes
                return sr.ReadToEnd();
        }
```
The name has the correct string: Serene.Web.Migrations.NorthwindDB.NorthwindDBScript_SqlServer.sql

It looks like it cannot find or load the SQL script into the streamreader. Question is why.

This is a completely fresh install with fresh checkout of newest serene .net core.

Any help is appreciated.

With kind regards,

John

**Comment by edwardch** [comment]
Hi John,

The Serenity apps I've run in .NET Core (one OSX, one linux Docker container) did not have any .sql scripts in the Migrations folder, so its hard to say...

I understand this is a new project, but perhaps check if the SQL script's `Build Action` is set to `Embedded Resource`. I haven't checked how to do this on something other than Visual Studio so you might need to search around. 

Hope this helps!

**Comment by JohnRanger** [comment]
Thanks @edwardch ,

I just have now removed all references to northwind according to the SF guide and now at least it starts kestrel up.

But I only get now a 2cm high - grey -login panel with no content or buttons at all.

As I am no Linux or .Net core expert, my evaluation of Serenity-based projects on Linux in the current status (2019-11-05) has surely only minimal relevance - but this is far from an out-of-the-box experience - unlike the SF on Windows with "real" .net.

So I will stop my experiment for now with SF on Lx and will try in 2-3 years from now again :-)

With kind regards,

John

**Comment by edwardch** [comment]
Hi John,

I recently set up a new .NET Core project with _no_ reference to Northwind. Then I copied some migrations and modules from an existing .NET MVC Serenity project into the Core project. There were a few `.sql` scripts in the migration folder that were working in the .NET MVC project.

I ran into a similar issue to what you explained above and couldn't figure out how to get it working. The file names were correct like in your case. 

As a workaround I ended up taking the SQL code out and put it into new migrations inside `Execute.Sql("")` methods and then the project ran fine.

I don't have any urgent `.sql` scripts to add to any projects, but if I do need to and I figure out how I will be sure to post an issue/tutorial/other to let you and anyone else know.

Cheers!
Edward
